### ISSUE TITLE
Better Support for Android TV

### ISSUE THREAD

**Issue by oakesville** [oakesville/mythling#46]
Mythling works nicely on FireTV, but has known issues Android TV that are reflected in the Google Play console ANR and Crashes reports.  I suspect that places in the code that have conditional logic based on AppSettings.isFireTv() need to be generalized to use AppSettings.isTv().  Or maybe isTv() isn't correctly identifying Android TV devices.  In any case, without an Android TV I'm unable to test and fix these issues.

**Comment by cannino** [comment]
So, I have started looking into what existing issues there are. I have a development environment running, but am having a hard time figuring out why menu options aren't shown on the upper right. 
I set a backend IP in the setup, but every launch of the app brings me to the recordings list without any menu items (recordings[#], by date, layout, ...) so I can't change to 'live tv' or other catagories.  Where in the source code can I debug what's going on? I tried reverse-engineering from the "Live TV" resource in Localizer.java but still can't find what controls the sub-menu rendering.

**Comment by oakesville** [comment]
Another problem recently reported on Shield TV:
```
Mythling does not respond to the play/pause button (the select button) on the shield tv remote.  There is, therefore no way to pause using the remote.  It works in plex, but not mythling.
```

**Comment by oakesville** [comment]
@cannino, The menu is driven by res/menus/main.xml.  You may be able to expose the menus by hitting a Settings or Options button on your remote if there is such a thing.  It sounds like maybe the main menu for Android TV should be more like firetv_main.xml (`android:showAsAction="always"`).  This is an example of what I mean when I say we should use the generic `isTv()` method instead of `isFireTv()`.  See `MediaActivity.onCreateOptionsMenu()`.  Hopefully `isTv()` returns true on your TV.

**Comment by cannino** [comment]
Well, at least I was looking in the right places. From what I have read the only supported controls AndroidTV allows are the d-pad so a settings or options doesn't exist.  ```isTV()``` does return true so that's good but I think we may be getting snagged by:
https://www.reddit.com/r/androiddev/comments/5ochh7/psa_oncreateoptionsmenu_is_not_called_on_android/
and
https://issuetracker.google.com/issues/37122103

It seems like this is exactly the behavior I am seeing. I think I am going to try and update the theme based on these links.

**Comment by cannino** [comment]
<img width="512" alt="image" src="https://user-images.githubusercontent.com/31353535/34063988-0763aac6-e1b3-11e7-9698-1b5dca1ec292.png">
It's working!
I have updated the theme and now see the actionbar. I was also able to play LiveTV! There are still a few kinks to work out (Crash when trying to play a recording) but it definitely is a lot further along than when I started.

I'll debug further on playing a recording but here's the stack trace in case it looks familiar to anyone:                   
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.oakesville.mythling/com.oakesville.mythling.MediaListActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.app.ActionBar.setDisplayHomeAsUpEnabled(boolean)' on a null object reference
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2707)
                      at android.app.ActivityThread.-wrap12(ActivityThread.java)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1460)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:154)
                      at android.app.ActivityThread.main(ActivityThread.java:6077)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
                   Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.app.ActionBar.setDisplayHomeAsUpEnabled(boolean)' on a null object reference
                      at com.oakesville.mythling.MediaListActivity.onCreate(MediaListActivity.java:80)
```

**Comment by oakesville** [comment]
@cannino You may need to invoke `getSupportActionBar()` instead of `getActionBar()`:
https://stackoverflow.com/questions/28144657/android-error-attempt-to-invoke-virtual-method-void-android-app-actionbar-on

**Comment by cannino** [comment]
Hi Don,
  I am getting stumped on using the libvlc player to play recordings. I am getting various UnsatisfiedLinkError's the latest being:
W/VLC/LibVLC: Unable to load the anw library: java.lang.UnsatisfiedLinkError: dlopen failed: library "libhardware.so" not found...couldn't find "libvlc.so"

When I extract the libvlc.aar you sent me, the armeabi-v7a folder gets copied to the libs folder correct? Here's the list of files in that directory: 
gdb.setup      libanw.10.so   libanw.14.so   libanw.21.so   libiomx.13.so  libvlcjni.so
gdbserver      libanw.13.so   libanw.18.so   libiomx.10.so  libiomx.14.so
I tried compiling vlc-android but ran into issues running the compile-libvlc.sh script during the 'buildsystem tools: make' phase. Any direction here would be greatly appreciated.

**Comment by cannino** [comment]
@oakesville Ready to push a PR, but looks like i may not have permissions. I have been able to play live tv as well as recordings using the Android Player. Could you add me as a contributor to the repository?

**Comment by oakesville** [comment]
@cannino I merged your PR, and opened a issue #94 for the ActionBar problems, which will need fixing before I publish a new release.  Thanks for your help.
